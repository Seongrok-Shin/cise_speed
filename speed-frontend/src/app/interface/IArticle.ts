interface IArticle {
    id: string;
    title: string;
    authors: string[];
    journal: string;
    year: number;
    volume: number;
    pages: number;
    DOI: string;
    status: string;
    claim: string;
    result: boolean;
    evidence: string;
    research: string;
    participant: string;
    se_practice: string;
    date: string;
    is_approved: {
      isModerator: boolean;
      isModRejected: boolean;
      isAnalyst: boolean;
      isAnaRejected: boolean;
    };
}

export default IArticle;
