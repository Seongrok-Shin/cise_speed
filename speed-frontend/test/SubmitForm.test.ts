import { describe, expect, test } from "@jest/globals";
import SubmitPageForm from "@/app/component/SubmitForm";
import DataInterface from "@/app/interface/IArticle";
import IArticle from "@/app/interface/IArticle";

describe("Form", () => {
  test("should match snapshot", () => {
    const handleChnage = (e: any) => {
      console.log(e);
    };
    const handleSubmit = (e: any) => {
      console.log(e);
    };
    const data: IArticle = {
      title: '',
    authors: [],
    journal: '',
    year: 0,
    volume: 0,
    pages: 0,
    DOI: '',
    status: '',
    claim: '',
    result: false,
    evidence: '',
    research: '',
    participant: '',
    se_practice: '',
    is_approved: {
      isModerator: false,
      isAnalyst: false,
      isAnaRejected: false,
      isModRejected: false,
    }
    };
    const form = SubmitPageForm(handleChnage, handleSubmit, data);
    expect(form).toMatchSnapshot();
  });
});
