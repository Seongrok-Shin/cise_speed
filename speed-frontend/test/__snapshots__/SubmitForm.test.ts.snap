// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<div
  className="flex flex-col items-center"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="py-2"
    >
      <input
        className="rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        required={true}
        type="text"
        value="d"
      />
    </div>
    <div>
      <input
        className="rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        name="author"
        onChange={[Function]}
        placeholder="Enter author"
        required={true}
        type="text"
        value={[]}
      />
    </div>
    <div
      className="py-2"
    >
      <input
        className="rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        name="journal"
        onChange={[Function]}
        placeholder="Enter journal"
        required={true}
        type="text"
        value="d"
      />
    </div>
    <div>
      <input
        className="w-full rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        name="year"
        onChange={[Function]}
        placeholder="Enter year"
        required={true}
        type="date"
        value="213"
      />
    </div>
    <div
      className="py-2"
    >
      <input
        className="rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        name="volume"
        onChange={[Function]}
        placeholder="Enter volume"
        required={true}
        type="text"
        value="123"
      />
    </div>
    <div>
      <input
        className="rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        data-testid="pages"
        name="pages"
        onChange={[Function]}
        placeholder="Enter pages"
        required={true}
        type="text"
        value="123"
      />
    </div>
    <div
      className="py-2"
    >
      <input
        className="rounded-xl border-2 border-gray-300 focus:outline-none focus:border-indigo-500 text-base px-4 py-2"
        name="DOI"
        onChange={[Function]}
        placeholder="Enter DOI"
        required={true}
        type="text"
        value="123"
      />
    </div>
    <button
      className="w-full rounded-xl border-2 border-gray-300 focus:outline-none   focus:border-indigo-500 text-base px-4 py-2"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
